plugin/evm/gossip_mempool.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ava-labs/coreth/plugin/evm/message"
+
+	"github.com/ava-labs/avalanchego/ids"
+	bloomfilter "github.com/holiman/bloomfilter/v2"
+
+	"github.com/ava-labs/coreth/core"
+	"github.com/ava-labs/coreth/core/txpool"
+	"github.com/ava-labs/coreth/core/types"
+	"github.com/ava-labs/coreth/gossip"
+)
+
+var (
+	_ gossip.Mempool[*GossipAtomicTx] = (*GossipAtomicTxMempool)(nil)
+	_ gossip.Tx                       = (*GossipAtomicTx)(nil)
+
+	_ gossip.Mempool[*GossipEthTx] = (*GossipEthTxMempool)(nil)
+	_ gossip.Tx                    = (*GossipEthTx)(nil)
+)
+
+func NewGossipAtomicTxMempool(Mempool *Mempool) (*GossipAtomicTxMempool, error) {
+	bloom, err := bloomfilter.New(gossip.DefaultBloomM, gossip.DefaultBloomK)
```

Comment:
1. I feel indifferent about documenting it personally, it's a hard-coded value that we probably won't expose as config
2. Hmm if someone wants to change this parameter I'd say it's dangerous - changing the parameters of the bloom filter is only safe if every other node has the same parameters you're using so it would only be useful in the context of a subnet-evm for example, otherwise you run the risk of the bloom filter formats being different between nodes. I think it's fine to leave it as is where it's hard-coded and if you want to change it you have to modify the source.

Reply:
Hmm I think hard-coded is fine for now. The danger is that this config is only safe to change given that all nodes in a subnet are using the same values and it's going to be very difficult for a user to debug why their transactions aren't getting gossiped if they shoot themselves in the foot and one node has a different bloom filter parameter, even if it was documented.